internal/pkg/cli/pipeline_init.go

```diff
@@ -0,0 +1,153 @@
+	manifestWriter archer.ManifestIO
+	prompt         prompter
+
+	// Outputs stored on successful actions.
+	manifestPath string
+}
+
+// Ask prompts for fields that are required but not passed in.
+func (opts *InitPipelineOpts) Ask() error {
+	// TODO do we need?
+	// if opts.AppName == "" {
+	// 	if err := opts.askAppName(); err != nil {
+	// 		return err
+	// 	}
+	// }
+	// if opts.GitHubRepo == "" {
+	// 	if err := opts.askGitHubRepo(); err != nil {
+	// 		return err
+	// 	}
+	// }
+	// if opts.GitHubAccessToken == "" {
+	// 	if err := opts.askGitHubAccessToken(); err != nil {
+	// 		return err
+	// 	}
+	// }
+
+	// if len(opts.Environments) == 0 {
+	// 	if err := opts.askEnvironments(); err != nil {
+	// 		return err
+	// 	}
+	// }
+	return nil
+}
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (opts *InitPipelineOpts) Validate() error {
+	// TODO
+	return nil
+}
+
+// Execute writes the pipline manifest file.
+func (opts *InitPipelineOpts) Execute() error {
+	// TODO
+	return nil
+}
+
+func (opts *InitPipelineOpts) askAppName() error {
+	// projectName := viper.GetString(projectFlag)
+	// apps, err := opts.appStore.ListApplications(projectName)
+	// if err != nil {
+	// 	return err
+	// }
+
+	// selections := make([]string, len(apps))
+	// for _, app := range apps {
+	// 	selections = append(selections, app.Name)
+	// }
+
+	// sel, err := opts.prompt.SelectOne(
+	// 	fmt.Sprintf("Which application would you like to create a pipeline for?"),
+	// 	`The application to create a pipeline for.`,
+	// 	selections,
+	// )
+	// if err != nil {
+	// 	return fmt.Errorf("failed to select application: %w", err)
+	// }
+
+	return nil
+}
+
+// BuildPipelineInitCmd build the command for creating a new pipeline.
+func BuildPipelineInitCmd() *cobra.Command {
+	opts := &InitPipelineOpts{}
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Creates a pipeline for applications in your workspace.",
+		Long: `Creates a pipeline for the applications in your workspace, using the environments associated with the applications."`,
+		Example: `
+  Create a pipeline for the applications in your workspace
+  /code $ archer pipeline init \
+    --github-repo "gitHubUserName/myFrontendApp" \
+    --github-access-token file://myGitHubToken \
+    --environments "stage,prod" \
+    --deploy`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.fs = &afero.Afero{Fs: afero.NewOsFs()}
+			opts.prompt = prompt.New()
+
+			store, err := ssm.NewStore()
+			if err != nil {
+				return fmt.Errorf("couldn't connect to project datastore: %w", err)
+			}
+			opts.appStore = store
+
+			ws, err := workspace.New()
+			if err != nil {
+				return fmt.Errorf("workspace cannot be created: %w", err)
+			}
+			opts.manifestWriter = ws
+
+			return opts.Validate()
+		},
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			log.Warningln("It's best to run this command in the root of your workspace.")
```

Comment:
We can probably get rid of this warning - only init is best to run from the WS root.